core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -1107,12 +1108,15 @@ public void postValidate(DownloadJob job, File src) throws IOException {
          */
         public File download(DownloadJob job, URL src) throws IOException {
             MessageDigest sha1 = null;
+            MessageDigest sha256 = null;
+            MessageDigest sha512 = null;
             try {
+                // Java spec says SHA-1 and SHA-256 exist, and SHA-512 might not, so one try/catch block should be fine
                 sha1 = MessageDigest.getInstance("SHA-1");
```

Comment:
I suggest to create a follow-up ticket to disable weak algorithms (e.g. via Security settings or system property)